### PR TITLE
feat: add bi-directional streaming dealer and router zmq clients

### PR DESCRIPTION
**Pull request by ajcasagrande**
This is for the upcoming Sticky Credit Router. It will allow us to send credits to specific workers.

## Release Notes

* **New Features**
  * Added streaming ROUTER and DEALER client types for bidirectional ZeroMQ communication with identity-based message routing and asynchronous message handlers.

* **Tests**
  * Added comprehensive test coverage for streaming clients, including initialization, message sending/receiving, lifecycle management, error handling, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/aiperf/zmq/streaming_dealer_client.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+            except zmq.Again:
+                self.debug("No data on dealer socket received, yielding to event loop")
+                await yield_to_event_loop()
+            except Exception as e:
+                self.exception(f"Exception receiving messages: {e}")
+                await yield_to_event_loop()
+            except asyncio.CancelledError:
+                self.debug("Streaming DEALER receiver task cancelled")
+                raise  # re-raise the cancelled error
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/zmq/streaming_dealer_client.py:185]: invalid. cancellederror is a BaseException and will not be caught.

File: src/aiperf/zmq/streaming_router_client.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                self.debug("Router receiver task timed out")
+                await yield_to_event_loop()
+                continue
+            except Exception as e:
+                if not self.stop_requested:
+                    self.exception(f"Error in streaming ROUTER receiver: {e}")
+                await yield_to_event_loop()
+            except asyncio.CancelledError:
+                self.debug("Streaming ROUTER receiver task cancelled")
+                break
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/zmq/streaming_router_client.py:204]: base exception. will not be caught

File: src/aiperf/zmq/streaming_dealer_client.py [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        except Exception as e:
+            self.exception(f"Failed to send message: {e}")
+            raise
+
+    @background_task(immediate=True, interval=None)
+    async def _streaming_dealer_receiver(self) -> None:
+        """
+        Background task for receiving messages from ROUTER.xz
+
+        Runs continuously until stop is requested. Receives messages with DEALER
```
Comment (ilana-n) [inline src/aiperf/zmq/streaming_dealer_client.py:156]: general q: will this use more CPU than the current implementation if we are running continuously in the background?

Comment (ajcasagrande) [reply src/aiperf/zmq/streaming_dealer_client.py:156]: @ilana-n no, this is what the current code does as well. Its async read so its basically sitting idle waiting for data to come in.
